csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -1849,60 +1849,62 @@ private void TestWeightSharingBetweenSessions()
             var dims = new long[] { 3, 2 };
             var dataBuffer = new float[] { 1.0F, 2.0F, 3.0F, 4.0F, 5.0F, 6.0F };
             var allocator = OrtAllocator.DefaultInstance;
-            var ortAllocationInput = allocator.Allocate((uint)dataBuffer.Length * sizeof(float));
-            unsafe
+            using (var ortAllocationInput = allocator.Allocate((uint)dataBuffer.Length * sizeof(float)))
             {
-                float* p = (float*)ortAllocationInput.DangerousGetHandle();
-                for (int i = 0; i < dataBuffer.Length; ++i)
+                unsafe
                 {
-                    *p++ = dataBuffer[i];
+                    float* p = (float*)ortAllocationInput.DangerousGetHandle();
```

Comment:
>DangerousGetHandle [](start = 58, length = 18)

Where did this come from? Why not Pointer?

Reply:
This is the code from the original PR.

Reply:
Pointer seems to be marked internal and it's doing the same thing as [DangerousGetHandle](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.safehandle.dangerousgethandle?view=netcore-3.1#System_Runtime_InteropServices_SafeHandle_DangerousGetHandle).

Reply:
I am afraid, it is not a good example. Can we make this test in the same manner as this: https://github.com/microsoft/onnxruntime/blob/master/csharp/test/Microsoft.ML.OnnxRuntime.Tests/OrtIoBindingAllocationTest.cs

Reply:
There are two reasons.  1) DangerouseGetHandle() is inherited from Safehandle and it is not a part of the official API we support. 2) There is a reason Pointer is internal and example of using Dangerous*() is allowing pointer leak outside.

Reply:
Resolved